rotkehlchen/chain/ethereum/modules/aave/graph.py

```diff
@@ -581,35 +655,65 @@ def _get_user_data(
             last_query_ts = last_query[1]
             from_ts = Timestamp(last_query_ts + 1)
 
-        deposits = withdrawals = borrows = repays = liquidation_calls = []
+        deposits, withdrawals, borrows = [], [], []
+        repays, liquidation_calls = [], []
         query = self.graph.query(
             querystr=USER_EVENTS_QUERY,
             param_types={'$address': 'ID!'},
             param_values={'address': address.lower()},
         )
-        user_result = query['users'][0]
+        query_v2 = self.graph_v2.query(
+            querystr=USER_EVENTS_QUERY_V2,
+            param_types={'$address': 'ID!'},
+            param_values={'address': address.lower()},
+        )
+
+        user_merged_data = defaultdict(list)
         if now - last_query_ts > AAVE_GRAPH_RECENT_SECS:
             # In theory if these were individual queries we should do them only if
             # we have not queried recently. In practise since we only do 1 query above
             # this is useless for now, but keeping the mechanism in case we change
             # the way we query the subgraph
-            deposits = self._parse_deposits(user_result['depositHistory'], from_ts, to_ts)
-            withdrawals = self._parse_withdrawals(
-                withdrawals=user_result['redeemUnderlyingHistory'],
-                from_ts=from_ts,
-                to_ts=to_ts,
-            )
-            borrows = self._parse_borrows(user_result['borrowHistory'], from_ts, to_ts)
-            repays = self._parse_repays(user_result['repayHistory'], from_ts, to_ts)
-            liquidation_calls = self._parse_liquidations(
-                user_result['liquidationCallHistory'],
-                from_ts,
-                to_ts,
-            )
+
+            if len(query['users']) != 0:
+                user_result = query['users'][0]
+                deposits += self._parse_deposits(user_result['depositHistory'], from_ts, to_ts)
+                withdrawals += self._parse_withdrawals(
+                    withdrawals=user_result['redeemUnderlyingHistory'],
+                    from_ts=from_ts,
+                    to_ts=to_ts,
+                )
+                borrows += self._parse_borrows(user_result['borrowHistory'], from_ts, to_ts)
+                repays += self._parse_repays(user_result['repayHistory'], from_ts, to_ts)
+                liquidation_calls += self._parse_liquidations(
+                    user_result['liquidationCallHistory'],
+                    from_ts,
+                    to_ts,
+                )
+                for key, value in user_result.items():
+                    user_merged_data[key].extend(value)
+
+            if len(query_v2['users']) != 0:
```

Comment:
These 2 ifs are identical. Create a function for it and then use it in both places, don't copy paste the exact same code.